### PR TITLE
Add JSONResponse

### LINKED ISSUE #592: Add json_response funciton
Should be derived from `aiohttp.web.Response`.

Constructor signature is:
`def __init__(self, data, *, status=200, reason=None, headers=None)`

Should pack `data` arg as `json.dumps()` and set content type to `application/json`.

People forget to specify proper content type on sending json data.

### PR DESCRIPTION
**Pull request by sloria**
closes #592 

Let me know if any further changes need to be made.

### CODE REVIEW

File: aiohttp/web_reqrep.py [hunk 1]
```diff
@@ -810,3 +812,20 @@ def write_eof(self):
         if body is not None:
             self.write(body)
         yield from super().write_eof()
+
+
+def json_dumps(data):
```
Comment (asvetlov) [inline aiohttp/web_reqrep.py]: I suggest 

```
class JSONResponse(Response):

    def __init__(self, data, *, body=None, status=200,
                 reason=None, headers=None, content_type='application/json',
                 dumps=json.dumps):
        super().__init__(text=text, status=status, reason=reason,
                         content_type=content_type)
```

File: aiohttp/web_reqrep.py [hunk 2]
```diff
@@ -810,3 +812,20 @@ def write_eof(self):
...
+
+
+class JSONResponse(Response):
+    content_type = 'application/json'
+
+    def __init__(self, data, *, body=None, status=200,
+                 reason=None, headers=None, content_type=None,
+                 dumps=json_dumps):
+        self.data = data
+        body = dumps(self.data)
```
Comment (jashandeep-sohi) [inline aiohttp/web_reqrep.py]: What happens when both `data` & `body` are specified?

**Comment by asvetlov** [comment]
My former comment was wrong.
See updated proposal:

```
class JSONResponse(Response):

    def __init__(self, data=sentinel, *, text=None, body=None, status=200,
                 reason=None, headers=None, content_type='application/json',
                 dumps=json.dumps):
        if data is not sentinel and (text of body):
            raise ValueError('only the one of data, text or body should be specified')
        text = dumps(data)
        super().__init__(text=text, body=body, status=status, reason=reason,
                         content_type=content_type)
```

File: aiohttp/web_reqrep.py (hunk 2 above)
Comment (asvetlov) [inline aiohttp/web_reqrep.py]: Good point. See the updated proposal in comments

**Comment by sloria** [comment]
Changes made. Let me know if further changes are necessary or if I should squash the commits.

**Comment by sloria** [comment]
What should happen if none of `data`, `text`, or `body` are passed?

The current behavior is not ideal:

``` python
JSONResponse()  # TypeError: <object object at 0x10062e400> is not JSON serializable
```

I can raise an error:

``` python
        if not any([data is not sentinel, text, body]):
            raise ValueError(
                'one of data, text, or body must be specified'
            )
```

but I'm not sure if we want to allow setting any of those attributes after a `JSONResponse` is instantiated, in which case I can do the following instead:

``` python
        if data is not sentinel:
            text = dumps(data)
        super()...
```

**Comment by asvetlov** [comment]
I've merged but replaced `JSONResponse` class with `json_response` function as @popravich suggested.

**Pull request merged**